### PR TITLE
Lavaland Buterching and Suit Fixes

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Maybe these should be different PRs, but it was in the same file so oh well.

Fix: You can use the goliath plates on your hood/hardsuit helmet again

Fix: The explorer suit properly protects your body from cold/heat now.

Notfix: Goliath plates and Watcher diamonds are inside the mob, you butcher the mob to retrieve the stuff. I did this because muh immersions, but if people think it will just be obnoxious to stand there for a few seconds I guess it can be dropped. This only impacts the lavaland versions of the mobs, asteroid mobs die the same as always.

Legion hearts drop on the ground as usual because it'd be extra work to make the corpse it drops have the heart in it, and it's often important to grab them off the ground mid combat.

:cl:
rscadd: You now need to butcher goliaths and watchers on lavaland to receive their loot. Use the combat knives to do so.
/:cl:

### CODE REVIEW

**Comment by Bawhoppen** [comment]
neat

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 1]
```diff
@@ -648,7 +646,8 @@
 	throw_message = "does nothing to the tough hide of the"
 	pre_attack_icon = "goliath2"
 	loot = list(/obj/item/asteroid/goliath_hide{layer = 4.1})
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: good job

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: oh no

**Comment by Core0verload** [comment]
How about linking hood/helmet upgrades to upgrades of suit? They are a single item (well, sort of), it would make sense that upgrades of one will affect another.

**Comment by Fox-McCloud** [comment]
@KorPhaeron 

While you're fixing mining stuff...your mining bot upgrades reference themselves when you upgrade things

"the combat card cannot be upgraded because it already has a module installed!"

Also, one of procs has `name` in it for some reason.

**Comment by bgobandit** [comment]
buterching

**Comment by xxalpha** [comment]
butterching

**Comment by lzimann** [comment]
this cold protection doesn't mean spaceproof, right?

**Comment by GunHog** [comment]
No, just as much as a firesuit.

**Comment by duncathan** [comment]
butter churn

**Pull request merged**